### PR TITLE
fix advanced payment builder message when there are insufficient funds

### PR DESCRIPTION
**Pull request by mmou**
Screen Shot 2020-04-05 at 2 11 15 PM
Screen Shot 2020-04-05 at 1 53 59 PM

### CODE REVIEW

**Comment by cecileboucheron** [comment]
Lgtm!

File: shared/wallets/send-form/asset-input/asset-input-advanced.tsx [hunk 1]
```diff
@@ -396,6 +398,7 @@ const styles = Styles.styleSheetCreate(
       },
       error: {
         color: Styles.globalColors.redDark,
+        maxWidth: '80%',
```
Comment (thebearjew) [inline shared/wallets/send-form/asset-input/asset-input-advanced.tsx:401]: 🤔There might be another way around this. I'll take a peek.

Comment (mmou) [reply shared/wallets/send-form/asset-input/asset-input-advanced.tsx:401]: lol

File: go/stellar/stellarsvc/frontend.go [hunk 2]
```diff
@@ -1216,7 +1216,7 @@ func (s *Server) FindPaymentPathLocal(ctx context.Context, arg stellar1.FindPaym
 		if err != nil {
 			return stellar1.PaymentPathLocal{}, err
 		}
-		res.AmountError = fmt.Sprintf("You only have %s available to spend.", availableToSpend)
+		res.AmountError = fmt.Sprintf("Your XLM balance is not sufficient. You only have %s available to spend.", availableToSpend)
```
Comment (songgao) [inline go/stellar/stellarsvc/frontend.go:1219]: The source asset could be something other than XLM. Is this function only used for XLMs?

File: shared/provision/select-other-device/index.tsx [hunk 3]
```diff
@@ -78,7 +78,7 @@ class SelectOtherDevice extends React.Component<Props> {
         title={
           this.props.passwordRecovery
             ? 'Recover password'
-            : `Authorize this ${Styles.isMobile ? 'phone' : 'computer'}`
+            : `Authorize this ${Styles.isMobile ? 'device' : 'computer'}`
```
Comment (mmou) [inline shared/provision/select-other-device/index.tsx:81]: finishing what https://github.com/keybase/client/commit/bc6648ac09b415b0c4c9041b77fa0abf0317e2eb started

File: shared/stories/__tests__/__snapshots__/Storyshots.test.js.snap [hunk 4]
```diff
@@ -531780,6 +531780,7 @@ Array [
                           "borderRadius": 4,
                           "borderStyle": "solid",
                           "borderWidth": 0,
+                          "flexShrink": 1,
```
Comment (mmou) [inline shared/stories/__tests__/__snapshots__/Storyshots.test.js.snap:531783]: ¯\_(ツ)_/¯

File: shared/wallets/send-form/asset-input/asset-input-advanced.tsx [hunk 5]
```diff
@@ -327,41 +328,39 @@ const PickAssetButton = (props: PickAssetButtonProps) => {
...
-      <Kb.Box style={styles.pickAssetButtonOverlayInner}>
-        <Kb.Box2
-          direction="vertical"
-          fullHeight={true}
-          alignSelf="flex-start"
-          alignItems="flex-end"
-          style={styles.pickAssetButton}
+      <Kb.Box2
+        direction="vertical"
+        fullHeight={false}
```
Comment (mmou) [inline shared/wallets/send-form/asset-input/asset-input-advanced.tsx:333]: needed to change this

File: shared/wallets/send-form/asset-input/asset-input-advanced.tsx [hunk 6]
```diff
@@ -469,22 +469,25 @@ const styles = Styles.styleSheetCreate(
         flexShrink: 0,
       },
       pickAssetButton: Styles.platformStyles({
-        common: {
-          width: Styles.globalMargins.xlarge * 3,
-        },
         isMobile: {
+          //    width: Styles.globalMargins.xlarge * 3,
```
Comment (mmou) [inline shared/wallets/send-form/asset-input/asset-input-advanced.tsx:473]: this line wasn't having any affect

File: shared/wallets/send-form/asset-input/asset-input-advanced.tsx [hunk 7]
```diff
@@ -469,22 +469,25 @@ const styles = Styles.styleSheetCreate(
...
           //    paddingTop: Styles.globalMargins.tiny,
         },
       }),
       // We need this to make the PickAssetButton on top of other stuff so amount
       // error can extend below it.
```
Comment (thebearjew) [inline shared/wallets/send-form/asset-input/asset-input-advanced.tsx:478]: @mmou  this comment is outdated, feel free to remove

File: shared/wallets/send-form/asset-input/asset-input-advanced.tsx [hunk 8]
```diff
@@ -469,22 +469,25 @@ const styles = Styles.styleSheetCreate(
...
-      pickAssetButtonOverlayInner: {position: 'absolute', right: 0, top: 0},
-      pickAssetButtonOverlayOuter: {position: 'relative'},
+      pickAssetButtonOverlayOuter: {
+        flexShrink: 0,
+        position: 'relative',
```
Comment (thebearjew) [inline shared/wallets/send-form/asset-input/asset-input-advanced.tsx:481]: Can remove this I believe

Comment (mmou) [reply shared/wallets/send-form/asset-input/asset-input-advanced.tsx:481]: just checking did you test with this removed?

Comment (thebearjew) [reply shared/wallets/send-form/asset-input/asset-input-advanced.tsx:481]: Yeah I toggled it off in the devtools and it was fine. We left it as `position: "relative"` after removing the absolute positioning, so it was extra. Nothing is positioned absolutely anymore.

**Pull request merged**